src/diet.ts

```diff
@@ -612,37 +635,56 @@ export function runDiet(): void {
             }
           }
           consumeSafe(countToConsume, menuItem.item);
-        } else if (menuItem.item === Mayo.flex) {
-          if (menuItems[menuItems.length - 1].item === $item`jumping horseradish`) {
-            MayoClinic.setMayoMinder(Mayo.zapine, countToConsume);
-          } else {
-            MayoClinic.setMayoMinder(Mayo.flex, countToConsume);
-          }
+        } else if ([Mayo.flex, Mayo.zapine].includes(menuItem.item)) {
+          MayoClinic.setMayoMinder(menuItem.item);
         } else if (menuItem.item === $item`pocket wish`) {
           acquire(1, $item`pocket wish`, 60000);
-          cliExecute(`genie effect ${menuItem.wishEffect}`);
+          cliExecute(`genie effect ${menuItem.effect}`);
+        } else if (menuItem.item === $item`Rethinking Candy`) {
+          synthesize(countToConsume, $effect`Synthesis: Greed`);
         } else if (menuItem.item !== $item`Special Seasoning`) {
-          consumeSafe(countToConsume, menuItem.item);
+          consumeSafe(countToConsume, menuItem.item, menuItem.additionalValue);
         }
       }
-      itemCount[1] -= countToConsume;
+      dietEntry.quantity -= countToConsume;
     }
-
-    fillSomeSpleen(diet);
   }
+}
 
-  // FIXME: More accurate decision on whether to use mojo filters.
-  const mojoFilterCount = 3 - get("currentMojoFilters");
-  let spleenValue = valuePerSpleen(bestSpleenItem);
-  if (haveEffect($effect`Synthesis: Greed`) < estimatedTurns()) {
-    // Estimate 3000 meat candy price per synth.
-    spleenValue = Math.max(spleenValue, 3 * 30 * baseMeat - 3000);
+export function runDiet(): void {
+  if (myFamiliar() === $familiar`Stooper`) {
+    useFamiliar($familiar`none`);
   }
-  if (mallPrice($item`mojo filter`) < spleenValue) {
-    acquire(mojoFilterCount, $item`mojo filter`, spleenValue, false);
-    if (have($item`mojo filter`)) {
-      use(Math.min(mojoFilterCount, availableAmount($item`mojo filter`)), $item`mojo filter`);
-      fillSomeSpleen(diet);
+
+  const dietBuilder = computeDiet();
+
+  if (globalOptions.simulateDiet) {
+    if (!get("_mimeArmyShotglassUsed")) {
+      printDiet(dietBuilder.shotglass());
     }
+
+    print("===== SIMULATED DIET =====");
+    printDiet(dietBuilder.diet());
+  } else {
+    pillCheck();
+
+    nonOrganAdventures();
+
+    if (have($item`astral six-pack`)) {
+      use($item`astral six-pack`);
+    }
+    if (!get("_mimeArmyShotglassUsed")) {
+      consumeDiet(dietBuilder.shotglass());
+    }
+
+    if (
+      get("barrelShrineUnlocked") &&
+      !get("_barrelPrayer") &&
+      $classes`Turtle Tamer, Accordion Thief`.includes(myClass())
+    ) {
+      cliExecute("barrelprayer buff");
```

Comment:
dumb question: is it worth checking to see if we actually plan on using enough fullness in the associated organ before pulling the trigger on this?